################################################################################
SHORT DESCRIPTION: 
################################################################################
Port git://xenbits.xen.org/xen.git:a8ceaf5c72e7cd80a393622d047bb17ed7ffab65

################################################################################
LONG DESCRIPTION: 
################################################################################
commit a8ceaf5c72e7cd80a393622d047bb17ed7ffab65
Author: Frediano Ziglio <frediano.ziglio@citrix.com>
Date:   Mon May 7 13:37:47 2012 +0100

    vgabios: Report mode not supported getting mode informations
    
    If you try to get mode information for an unsupported mode
    interrupt should return error but not that the function is not
    supported.
    
    Signed-off-by: Frediano Ziglio <frediano.ziglio@citrix.com>
    Committed-by: Keir Fraser <keir@xen.org>

################################################################################
CHANGELOG 
################################################################################
Ported to VGABIOS: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015

################################################################################
REMOVAL 
################################################################################
This patch can be removed if a new version of Bochs VGABIOS is released with
this fix.

################################################################################
UPSTREAM PLAN 
################################################################################
This is already upstream in Xen repository.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES
################################################################################
--- a/vbe.c
+++ b/vbe.c
@@ -921,7 +921,8 @@ Bit16u *AX;Bit16u ES;Bit16u DI;
 void vbe_biosfn_return_mode_information(AX, CX, ES, DI)
 Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u DI;
 {
-        Bit16u            result=0x0100;
+        // error by default is 0x014f which means supported but error
+        Bit16u                 result=0x014f;
         Bit16u            ss=get_SS();
         ModeInfoBlock     info;
         ModeInfoListItem  *cur_info;
@@ -968,7 +969,6 @@ Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u D
 #ifdef DEBUG
                 printf("VBE *NOT* found mode %x\n",CX);
 #endif
-                result = 0x100;
         }
 
         if (result == 0x4f)
